reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/TakeObservable.kt

```diff
@@ -0,0 +1,43 @@
+package com.badoo.reaktive.observable
+
+import com.badoo.reaktive.base.subscribeSafe
+import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.disposable.DisposableWrapper
+
+/**
+ * Emit only the first [limit] items emitted by source.
+ */
+fun <T> Observable<T>.take(limit: Int): Observable<T?> {
+    require(limit > 0) { "count >= 0 required but it was $limit" }
```

Comment:
count **>** 0 required but it was $limit

Reply:
I would say 0 should be a valid value, so `require(limit >= 0)`.

Reply:
Can you think of a valid usecase for allowing `take(0)`? Isn't that as good as completing/disposing upstream right away? 

I actually copied this check from RxJava.

Reply:
I could imagine one, if limit is calculated somehow or received from e.g. server, like:
`users.take(server.getUserLimit()).otherOperators()`
Here 0 might be a valid value, so you don't need to care about it. If 0 would not be allowed, then you will have to write something like:
```
users
    .let { source ->
        server
            .getUserLimit()
            .takeIf { it > 0 }
            ?.let(source::limit) 
            ?: source
    }
    .otherOperators()
```

Also 0 is allowed in stdlib functions like `Iterable<T>.take(n)`. Also RxJava2's `Observable.take` allows 0.